src/Client/OAuth2PKCEClient.php

```diff
@@ -45,13 +45,14 @@ public function __construct(AbstractProvider $provider, RequestStack $requestSta
      */
     public function redirect(array $scopes = [], array $options = [])
     {
-        $this->getSession()->set(static::VERIFIER_KEY, $code_verifier = bin2hex(random_bytes(64)));
-        $pkce = [
-            'code_challenge' => rtrim(strtr(base64_encode(hash('sha256', $code_verifier, true)), '+/', '-_'), '='),
-            'code_challenge_method' => 'S256',
-        ];
+        $response = parent::redirect($scopes, $options);
 
-        return parent::redirect($scopes, $options + $pkce);
+        $codeVerifier = $this->getOAuth2Provider()->getPkceCode();
```

Comment:
I suppose this is the new method upstream that was not available on the moment when this client was written.

UPD: Actually, I see it was released on the latest 2.7 of league/oauth2-client, right? So, we literally pushed this dependency to the latest here. But it might be ok, at least tests are happy about it